### DISCUSSION TITLE
Redesigning the User interface to be simple, minimal and useful.

### DISCUSSION THREAD

**Discussion by setanarut** [sbarex/QLMarkdown#14] [Show and tell]
I don't like the UI design, There are many unnecessary bloatware features (syntax palette editor) and options (tabs, checkboxes) that do not concern the end user, I don't understand the purpose of having two folders named **theme** and **style**. this is a [category mistake](https://en.wikipedia.org/wiki/Category_mistake). All settings related to theme appearance should be stored in a single theme file (syntax highlighting palette included). Only a few important Parser settings (extensions, css theme selection, import-export css, markdown preview) should be determined with the UI, all settings related to the theme should be removed from the interface(syntax color editor. source code tab)

## My Minimalist UI design;

<img width="835" alt="redesign1" src="https://user-images.githubusercontent.com/36481442/103173281-f17a7600-486a-11eb-828f-a764619870c8.png">

<img width="835" alt="redesign2" src="https://user-images.githubusercontent.com/36481442/103173285-fb9c7480-486a-11eb-9a1d-3aa9e8825680.png">

## App Menu

```
QLMarkdown
	About
	Import Theme
	Check for Updates
	Quit
```

## Notes

 - .theme files should include information about theme name, theme creator name, etc.. and a "CSS" file with syntax color palette.
- These source code settings must be defined and managed in central default.theme file. Should be removed from the user interface.
- Too many unnecessary default highlight options, 10 most famous ones should be provided by default. there is no need for this space-taking gallery. These options must be included in the .theme file;

**Comment by setanarut** [comment]
"Style" menu label should be corrected as "Theme" I typed incorrectly.

**Comment by sbarex** [comment]
Thanks for the suggestions, I agree that the interface is a bit heavy with elements, but I prefer to have all possible customizations rather than settle for the choice made by the programmer. 
Furthermore, after configuring the first time, the main application will probably no longer open.

The **theme** folder is for the source language .theme files, each theme have only one appearance (light/dark).
The **style** folder is for custom .css files (valid for the base markdown document, but can override the source highlight), each file must handle light/dark appearance.

Default style for now do not have a name because is the only css provided. Also it is a file founded somewere on internet (I do not remember where).

**Comment by setanarut** [comment]
"I prefer to have all possible customizations rather than settle for the choice made by the programmer."

If you want an editor for those who can't write CSS, you can open a separate Theme Editor tab or Window. My ideas are easier for beginners. Although I am an experienced person, I had difficulty understanding your interface.

**Comment by relikd** [comment]
I would go for a simple plugin solution and remove the user editing capability. A zip-like file that contains all theme and/or styling options in itself. Then there could be a small showroom to preview and download the themes. This does not have to be part of the app but could be a web service. And the app is distributed with just a single (default) theme. I would choose the Github style as the baseline. Thats just my 2 cents.

It would also reduce the maintanence effort. For example, your theme preview breaks as soon as you choose "C source code (.c)" instead of the default "Theme colors". For some its completely blank, for others it uses the serif font again, weird colors, and line numbering. All in all keeping the app consistent gets harder with more settings that have a preview.

As to the other options. I really dont see that Emoji and Strikethrough should be a separate tab.

Also, I am constantly pressing CMD-W to try to close windows – which doesn't work.

**Reply by relikd** [reply]
e.g. something similar to this live demo https://github.com/jasonm23/markdown-css-themes (just a little nicer looking :D)

**Comment by sbarex** [comment]
CMD-w fixed on the last commit.

**Comment by sbarex** [comment]
Release 1.b6 has some little ui improvements.

**Reply by relikd** [reply]
Much better already. Can you merge the Icon Toolbar and the Tabbed View Controller? Either in one or the other. I dont think it makes much difference where but I think I'd prefer the icon toolbar.

__EDIT:__ or maybe a side-panel?

**Reply by setanarut** [reply]
better but not enough. I still believe there should be only single theme file. 

I want to write a few mistakes I noticed;

Wrong label name `Themes`. Should be `Syntax Highlight Editor` or similar;

<img width="57" alt="Ekran Resmi 2020-12-27 21 15 37" src="https://user-images.githubusercontent.com/36481442/103177165-0f56d380-4889-11eb-88a2-42a03a1620b1.png">

these options will conflict with CSS variables.
<img width="501" alt="Ekran Resmi 2020-12-27 21 06 48" src="https://user-images.githubusercontent.com/36481442/103177036-a458cd00-4887-11eb-8927-111557b48b77.png">


"Edit syntax highlight theme colors"
<img width="281" alt="Ekran Resmi 2020-12-27 21 05 54" src="https://user-images.githubusercontent.com/36481442/103177112-96f01280-4888-11eb-89b6-6d5683137df8.png">


Markdown color?? 😕 
<img width="200" alt="Ekran Resmi 2020-12-27 21 11 51" src="https://user-images.githubusercontent.com/36481442/103177305-45e11e00-488a-11eb-9075-96fd9a5b1b57.png">

**Comment by setanarut** [comment]
@sbarex can you make refresh shortcut more accessible like only "R". and Automatic refresh option on every change would be nice.

**Reply by sbarex** [reply]
> @sbarex can you make refresh shortcut more accessible like only "R". and Automatic refresh option on every change would be nice.

In the next release I have impletemented the autorefresh and the shortcut cmd-r for refresh.

**Comment by setanarut** [comment]
icon that should be dark looks light

<img width="94" alt="Ekran Resmi 2020-12-28 17 30 17" src="https://user-images.githubusercontent.com/36481442/103222328-3a96fc80-4935-11eb-9e5a-5348f7db4fe0.png">

**Comment by setanarut** [comment]
this section should be `monospace` font

<img width="503" alt="Ekran Resmi 2020-12-28 17 52 55" src="https://user-images.githubusercontent.com/36481442/103222650-9a8da300-4935-11eb-9e01-680b4747841f.png">

**Comment by setanarut** [comment]
@sbarex It will be clearer if you rename the UI items as follows;

- In Application Support folder:
  style -> themes
  theme -> syntax-highlight-color-schemes
-  "Document style" menu
  Document style -> Theme
  Default style -> Github ( Default )
  <img width="390" alt="proposal" src="https://user-images.githubusercontent.com/36481442/103470427-3595db00-4d83-11eb-898e-4afae7831984.png">

## Terminology for all labels and menu items etc;

Source Code Highlight -> Syntax Highlighting
Source code style - > Syntax Color Scheme
document style -> Theme
Inherit from document style -> Inherit from theme
Choose a source code style... -> Choose from gallery
Source Code Styles -> Syntax Color Schemes
Background -> Background color
From source code style -> From syntax color scheme
From document style -> From theme

open the syles editor -> Open color scheme editor
QLMarkdown - Source code highlight styles -> Syntax Color Scheme Editor

**Comment by sbarex** [comment]
I have implemented most of the suggestions in release 1.0b14.

**Reply by setanarut** [reply]
thanks

**Comment by setanarut** [comment]
Also;

`markdown.css` -> `default.css` or `github.css`


markdown -> Markdown
macos -> macOS
<img width="306" alt="Ekran Resmi 2021-01-04 00 46 35" src="https://user-images.githubusercontent.com/36481442/103489568-89a5cb80-4e26-11eb-9291-84b3dc8c5dbc.png">

**Comment by setanarut** [comment]
<img width="306" alt="Ekran Resmi 2021-01-04 02 42 23" src="https://user-images.githubusercontent.com/36481442/103491661-b235c180-4e36-11eb-8ef8-24ab6a27fd56.png">

quicklook -> Quick Look

**Reply by relikd** [reply]
also macOS

**Comment by sbarex** [comment]
fixed, thanks
